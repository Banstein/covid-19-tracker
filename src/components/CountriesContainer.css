.header-countries {
  max-height: 100px;
  background-color: var(--primary);
  padding: 5px;
}

.world {
  width: 150px;
  background-color: var(--primary);
}

.map {
  width: auto;
  height: 50px;
  background-color: var(--secondary);
}

.title {
  font-size: 40px;
  text-align: center;
}

.stats {
  padding: 5px;
}

.search-input {
  margin: 0 5px;
  padding: 0 5px;
  border: none;
  border-radius: 5px;
}
