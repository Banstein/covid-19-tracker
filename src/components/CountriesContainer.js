/* eslint-disable camelcase */
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import OneCountry from './OneCountry';
import './CountriesContainer.css';
import transformDate from '../helpers/transformDate';

function CountriesContainer() {
  const [searchField, setSearchField] = useState('');
  const countries = useSelector((state) => state.countriesReducer);
  const date = transformDate();

  const handleSearch = (e) => setSearchField((e.target.value).toLowerCase());
  const filteredCountries = countries.filter((country) => country.id.includes(searchField));

  return (
    <div>
      <div className="flex justify-evenly header-countries container">
        <img className="world" src="https://simplemaps.com/static/demos/resources/svg-library/svgs/world.svg" alt="" />
        <h1 className="title">WORLD</h1>
      </div>
      <div className="bg-secondary">
        <div className="flex container stats justify-between">
          <h5 className="title-stats">
            Stats by country
            {' '}
            {date}
          </h5>
          <input
            type="text"
            name="search"
            id="search"
            placeholder="Search Countries"
            value={searchField}
            onChange={handleSearch}
            className="search-input"
          />
        </div>

      </div>
      <ul className="ul-grid container">
        {filteredCountries.map(({ id, name, today_confirmed }) => (
          <OneCountry
            key={id}
            id={id}
            name={name}
            today_confirmed={today_confirmed}
            className="striped"
          />
        ))}
      </ul>
    </div>
  );
}

export default CountriesContainer;
