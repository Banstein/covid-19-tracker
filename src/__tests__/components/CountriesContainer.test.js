import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../__mocks__/configureStore';
import CountriesContainer from '../../components/CountriesContainer';

describe('Countries snapshot', () => {
  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <CountriesContainer />
        </BrowserRouter>
      </Provider>,
    );
  });
});
