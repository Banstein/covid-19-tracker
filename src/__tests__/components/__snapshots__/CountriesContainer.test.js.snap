// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<div>
  <div
    className="flex justify-evenly header-countries container"
  >
    <img
      alt=""
      className="world"
      src="https://simplemaps.com/static/demos/resources/svg-library/svgs/world.svg"
    />
    <h1
      className="title"
    >
      WORLD
    </h1>
  </div>
  <div
    className="bg-secondary"
  >
    <div
      className="flex container stats justify-between"
    >
      <h5
        className="title-stats"
      >
        Stats by country
         
        2022-04-15
      </h5>
      <input
        className="search-input"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search Countries"
        type="text"
        value=""
      />
    </div>
  </div>
  <ul
    className="ul-grid container"
  />
</div>
`;
